synergy-design-system/synergy-design-system
# chore: 🔧 setup code formatting rules for non TS/JS/CSS files

# Pull Request

## 📖 Description

This PR adds an initial prettier configuration for the whole monorepo. Currently, it is used to format all files that are not yet covered by another linter (e.g. eslint, stylelint) and operates on JSON and Markdown files primarly. 

### 🎫 Issues

Closes #181.

## 👩‍💻 Reviewer Notes

The PR holds various smaller changes:

- Basic prettier configuration (have a look at the current ignore rules!)
- Updated documentation
- Two new scripts that can be run in the root of the repo: `pnpm format.prettier` will just format the non code files, `pnpm format.eslint` will do the same for eslint. You can also use `pnpm format` to run them in parallel.
- > When #116 is tackled, we will remove `format.eslint`.
- To make sure the tokens don´t break, we have omitted the formatting of the `packages/tokens/src` folder.

## 📑 Test Plan

- Change a markdown or json file anywhere in the project, adding some unneeded whitespace.
- Run `pnpm format` (more complete) or `pnpm format.prettier` (faster) in the root
- The file should be reverted to its original state.

- Change a JSON file __in packages/tokens/src/figma-tokens` in the project, adding some unneeded whitespace.
- Run `pnpm format` (more complete) or `pnpm format.prettier` (faster) in the root
- The file should still look the same.
- Revert your manual changes :)

## ✅ DoD
